src/plugins/ssdtmon/ssdtmon.cpp

```diff
@@ -353,6 +423,40 @@ ssdtmon::ssdtmon(drakvuf_t drakvuf, const ssdtmon_config* config, output_format_
         if ( !drakvuf_add_trap(drakvuf, &this->ssdt_trap[3]) )
             throw -1;
     }
+
+    addr_t sdt_rva, sdt_shadow_rva;
+
+    if (!drakvuf_get_kernel_symbol_rva(drakvuf, "KeServiceDescriptorTable", &sdt_rva) ||
+        !drakvuf_get_kernel_symbol_rva(drakvuf, "KeServiceDescriptorTableShadow", &sdt_shadow_rva))
+    {
+        PRINT_DEBUG("[SSDTMON] Failed to get RVA of nt!KeServiceDescriptorTableShadow or nt!KeServiceDescriptorTable\n");
+        throw -1;
+    }
+
+    this->sdt_va = drakvuf_get_kernel_base(drakvuf) + sdt_rva;
+    this->sdt_shadow_va = drakvuf_get_kernel_base(drakvuf) + sdt_shadow_rva;
+
+    bool is64 = (drakvuf_get_page_mode(drakvuf) == VMI_PM_IA32E);
+    // SDT - 4 pointers long
+    this->sdt_crc = ssdtmon_sha256_calc(vmi, this->sdt_va, is64 ? 32 : 16);
+    // SDT shadow - 8 pointers long
+    this->sdt_shadow_crc = ssdtmon_sha256_calc(vmi, this->sdt_shadow_va, is64 ? 64 : 32);
+}
+
+bool ssdtmon::stop()
+{
+    if (!is_stopping() && !done_sdt_check)
+    {
+        m_is_stopping = true;
+        breakpoint_in_system_process_searcher bp;
+        if (!register_trap(nullptr, sdt_check_cb, bp.for_syscall_name("KiDeliverApc")))
```

Comment:
I'm still not a fan of adding traps when "stop" is being called..

Reply:
You can stash a copy of `drakvuf` in your own constructor if the pluginex copy is private. But tbh just make the `pluginex` copy public and be done with it.